querybook/webapp/hooks/queryEditor/extensions/useSqlCompleteExtension.ts

```diff
@@ -0,0 +1,87 @@
+import { inlineCopilot } from 'codemirror-copilot';
+import { useMemo } from 'react';
+
+import { AICommandType, AISocketEvent } from 'const/aiAssistant';
+import aiAssistantSocket from 'lib/ai-assistant/ai-assistant-socketio';
+
+/**
+ * Remove overlapping part with prefix from completion.
+ *
+ * e.g. removeOverlapPrefix('sel', 'elect *') => 'ect *'
+ */
+const removeOverlapPrefix = (completion: string, prefix: string) => {
```

Comment:
just curious why is this needed?

Reply:
It is a hack workaround. found sometimes it will have overlap with the prefix. I'll add some comments there.